### PR TITLE
Register new snapshots

### LINKED ISSUE #26602: msvc: Support a i686-pc-windows-msvc host compiler
We should provide nightlies for a i686-pc-windows-msvc host compiler (much in the same way we have a x86_64-pc-windows-msvc host compiler). Unfortunately there is currently a bug in LLVM preventing us from doing so, so the bootstrap aborts at libsyntax because of this bug: https://llvm.org/bugs/show_bug.cgi?id=23957

Build system support is being added in https://github.com/rust-lang/rust/pull/26601, but it's largely untested because a compiler could never be produced!

### PR DESCRIPTION
**Pull request by alexcrichton**
Enables bootstrapping a 32-bit MSVC host compiler!

Closes #26602

### CODE REVIEW

File: src/snapshots.txt [hunk 1]
```diff
@@ -1,3 +1,11 @@
+S 2015-07-17 d4432b3
+  linux-i386 93f6216a35d3bed3cedf244c9aff4cd716336bd9
+  linux-x86_64 d8f4967fc71a153c925faecf95a7feadf7e463a4
+  macos-i386 29852c4d4b5a851f16d627856a279cae5bf9bd01
+  macos-x86_64 1a20259899321062a0325edb1d22990f05d18708
+  winnt-i386 df50210f41db9a6f2968be5773b8e3bae32bb823
+  winnt-x86_64 d7774b724988485652781a804bdf8e05d28ead48
```
Comment (eefriedman) [inline src/snapshots.txt]: Is it intentional that this doesn't include snapshots for bitrig or freebsd?

Comment (alexcrichton) [inline src/snapshots.txt]: Currently, yes, I believe that @dhuseby is going to prepare some new snapshots soon.

**Comment by rillian** [comment]
```
$ curl -s https://static.rust-lang.org/dist/channel-rust-nightly
[...]
rust-nightly-i686-pc-windows-msvc.tar.gz
```
 \o/ Thanks Alex!

**Pull request merged**